scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -0,0 +1,384 @@
+import numpy as np
+from scipy.sparse.linalg import LinearOperator
+from scipy.sparse import spdiags, kron, eye, dia_array
+
+__all__ = ['LaplacianNd']
+
+
+class LaplacianNd(LinearOperator):
+    """
+    The grid Laplacian in `N` dimensions and its eigenvalues.
+
+    Construct Laplacian on a uniform rectangular grid in `N` dimensions
+    and output its eigenvalues. The Laplacian matrix `L` is square real
+    symmetric negative definite with signed interger entries and many zeros.
+
+    Parameters
+    ----------
+    shape : tuple
+        A tuple of integers of length `N` (corresponding to the dimension of
+        the Lapacian), where each entry gives the size of that dimension. The
+        Laplacian matrix is square of the size ``np.prod(shape)``
+    boundary_conditions: ``'Dirichlet'`` or ``'Neumann'`` or ``'Periodic'``
+        The type of the boundary conditions on the boundaries of the grid
+    dtype: type
+        The dtype of the matrix output. Default: ``np.int32``.
+
+    Attributes
+    ----------
+    :obj:`H <H>`
+        Hermitian adjoint. Implemented by the `LinearOpeator` class.
+    :obj:`T <T>`
+        Transpose this linear operator. Implemented by the `LinearOpeator` class.
+
+    Methods
+    -------
+    adjoint:
+        Implemented by the `LinearOpeator` class.
+    dot:
+        Implemented by the `LinearOpeator` class.
+    matmat: callable object
+        The handle to a function that multiplies the Laplacian matrix `L` of
+        the shape `n`-by-`n` on the right by an input matrix `x` of
+        the shape `n`-by-`k` to output ``L @ x`` without constructing `L`
+    matvec: callable object
+        The handle to a function that multiplies the Laplacian matrix `L` of
+        the shape `n`-by-`n` on the right by an input vector `x` of
+        the shape `n`-by-`1` to output ``L @ x`` without constructing `L`
+    rmatmat: callable object
+        The same as `matmat` since the Laplacian is real symmetric.
+    rmatvec: callable object
+        The same as `matmat` since the Laplacian is real symmetric.
+    transpose:
+        Implemented by the `LinearOpeator` class.
+    toarray: (n, n) ndarray
+        The ``(n, n)`` Laplacian matrix in a dense 2D array format
+    tosparse() : (n, n) sparse format
+        The ``(n, n)`` Laplacian matrix in a sparse format
+    eigenvalues : (n, ) ndarray, float
```

Comment:
eigenvalues is currently an attribute not method

Reply:
I was not sure how to handle it

Reply:
Ok I can have a look tomorrow

Reply:
I hope that I have now fixed it, but please double-check